core/profile.cpp

```diff
@@ -677,31 +677,28 @@ static void setup_gas_sensor_pressure(const struct dive *dive, const struct dive
 	std::vector<int> first(num_cyl, 0);
 	std::vector<int> last(num_cyl, INT_MAX);
 
-	int prev = dive->explicit_first_cylinder(dc);
-	prev = prev >= 0 ? prev : 0;
-	seen[prev] = 1;
-
-	event_loop loop("gaschange");
-	while (auto ev = loop.next(*dc)) {
-		int cyl = ev->gas.index;
-		int sec = ev->time.seconds;
+	int prev = -1;
+	gasmix_loop loop(*dive, *dc);
+	while (loop.has_next()) {
+		std::pair<int, int> gasmix_event = loop.next_cylinder_index();
 
-		if (cyl < 0)
+		if (gasmix_event.first < 0)
 			continue; // unknown cylinder
-		if (cyl >= num_cyl) {
-			report_info("setup_gas_sensor_pressure(): invalid cylinder idx %d", cyl);
+		if (gasmix_event.first >= num_cyl) {
+			report_info("setup_gas_sensor_pressure(): invalid cylinder idx %d", gasmix_event.first);
 			continue;
 		}
 
-		last[prev] = sec;
-		prev = cyl;
+		if (prev >= 0) {
+			last[prev] = gasmix_event.second;
 
-		last[cyl] = sec;
-		if (!seen[cyl]) {
-			// The end time may be updated by a subsequent cylinder change
-			first[cyl] = sec;
-			seen[cyl] = 1;
+			if (!seen[gasmix_event.first])
+				first[gasmix_event.first] = gasmix_event.second;
 		}
+
+		seen[gasmix_event.first] = 1;
```

Comment:
Since you are touching this line, you could use the more sensible `true` instead of `1`.

Reply:
This was used in some form of tri-state-ish. Refactored in [`b2e679d` (#4299)](https://github.com/subsurface/subsurface/pull/4299/commits/b2e679d34b1d133484f05d8fb425fcc6ea0aab37)